### ISSUE TITLE
Codesigning the libxlsxwriter framework fails on macOS

### ISSUE THREAD

**Issue by MrMage** [jmcnamara/libxlsxwriter#97]
When trying to depend on libxlsxwriter via CocoaPods in a macOS target, signing libxlsxwriter.framework fails with the following error message:

```
CodeSign /Users/daniel/Library/Developer/Xcode/DerivedData/libxslwriterCocoaPodsExample-egttyowdzjjhykaqqjqtjoicwuws/Build/Products/Debug/libxlsxwriter-OSX/xlsxwriter.framework/Versions/A
    cd /Users/daniel/Documents/Xcode/Mac/libxslwriterCocoaPodsExample/Pods
    export CODESIGN_ALLOCATE=/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/codesign_allocate
    
Signing Identity:     "-"

    /usr/bin/codesign --force --sign - --timestamp=none /Users/daniel/Library/Developer/Xcode/DerivedData/libxslwriterCocoaPodsExample-egttyowdzjjhykaqqjqtjoicwuws/Build/Products/Debug/libxlsxwriter-OSX/xlsxwriter.framework/Versions/A

/Users/daniel/Library/Developer/Xcode/DerivedData/libxslwriterCocoaPodsExample-egttyowdzjjhykaqqjqtjoicwuws/Build/Products/Debug/libxlsxwriter-OSX/xlsxwriter.framework/Versions/A: code object is not signed at all
In subcomponent: /Users/daniel/Library/Developer/Xcode/DerivedData/libxslwriterCocoaPodsExample-egttyowdzjjhykaqqjqtjoicwuws/Build/Products/Debug/libxlsxwriter-OSX/xlsxwriter.framework/Versions/A/xlsxwriter.h
```

The problem is that the framework target is set up such that `xlsxwriter.h` is put into the framework's main directory, alongside the `xlsxwriter` binary. It appears that the `codesign` utility only expects binaries in that directory when signing Mac frameworks, though, with all headers being in the `Headers` subdirectory (otherwise it fails with an `xlsxwriter.h: code object is not signed at all` message). Signing frameworks for iOS works fine, even if `xlsxwriter.h` is in the framework's main directory.

Here's a sample project to reproduce this issue: [libxslwriterCocoaPodsExample.zip](https://github.com/jmcnamara/libxlsxwriter/files/804903/libxslwriterCocoaPodsExample.zip).

The issue occurs both at `HEAD` and with the latest published CocoaPod.

https://github.com/jmcnamara/libxlsxwriter/pull/94 fixes this issue and introduces no regressions on iOS. To confirm the fix, simply change the two `https://github.com/jmcnamara/libxlsxwriter` URLs in the example's `Podfile` to `https://github.com/MrMage/libxlsxwriter`, run `pod install` and rebuild (when building for iOS, make sure to build for a device as simulator binaries are not codesigned anyway).

**Comment by jmcnamara** [comment]
@lrossi Could you have a look at this and the associated pull request. Thanks.

**Comment by lrossi** [comment]
Yes, I'll take a look ASAP

**Comment by dlpigpen** [comment]
@lrossi  Got the same issue. Can you fix it?

**Comment by jmcnamara** [comment]
@dlpigpen Does the pull request in #94 fix your issue?

**Comment by jmcnamara** [comment]
Closed by PR #127.
